### PR TITLE
feat: Added OpenAPI metadata to version docs (#38)

### LINKED ISSUE #38: Missing API Metadata in versioned docs
### Subject of the issue
I noticed that all [metadata of an API](https://fastapi.tiangolo.com/tutorial/metadata/#metadata-for-api) is displayed in the "main" documentation ("/swagger"). However, as soon as a versioned documentation ("/v1/swagger") is called, it is missing.

### Your environment
* 3.10
* Linux

### Steps to reproduce
overwrite `app` in [examples/simple.py](https://github.com/alexschimpf/fastapi-versionizer/blob/e7f75b4c1d40718afc71db1b8f7cb0a136a1e269/examples/simple.py)
```python
app = FastAPI(
    title='test',
    docs_url='/swagger',
    openapi_url='/api_schema.json',
    redoc_url=None,
    description='This is a very long description of my API.',
    summary='This is a short summary of my API.',
    contact={
        'name': 'SourceCode Github repository',
        'url': 'https://github.com/alexschimpf/fastapi-versionizer'
    },
    terms_of_service='https://example.com',
    license_info={
        'name': 'MIT',
        'url': 'https://opensource.org/licenses/MIT'
    },
)
```

### Expected behaviour
That's exactly the question why I'm creating an issue on github instead of making a pull request directly. I am aware of [the "Docs Customization" part in the readme](
https://github.com/alexschimpf/fastapi-versionizer/blob/e7f75b4c1d40718afc71db1b8f7cb0a136a1e269/README.md#docs-customization). I would assume, however, that in the normal case the metadata is transferred to all docs and I only have to touch the callback if I want to have different metadata in different versions.

**What is your opinion? I would be happy to create a pull request here so that all metadata is transferred by default.**

one more thing: If you agree to the above, should I also take over the "summary"? This attribute is only available since FastAPI 0.99.0 and according to your requirements.txt you still support down to 0.86.0.

### Actual behaviour
/swagger             |  /v1/swagger
:-------------------------:|:-------------------------:
grafik  |  grafik

### PR DESCRIPTION
**Pull request by worldworm**
## Pull Request Checklist

- [x] Make sure to read the contributor guide [here](https://github.com/alexschimpf/fastapi-versionizer/tree/main/CONTRIBUTE.md).
- [x] Make sure you are making a pull request against the main branch.
- [x] Make sure your pull request title matches the requested format.
- [x] Make sure to lint, type-check, and run unit and API tests.

## Description
closes #38

My first try looked like this:
```python
            async def get_openapi() -> Any:
                openapi_args = {
                    'title': title,
                    'version': version_str,
                    'routes': router.routes,
                    'description': self._app.description,
                    'terms_of_service': self._app.terms_of_service,
                    'contact': self._app.contact,
                    'license_info': self._app.license_info,
                }
                if hasattr(self._app, 'summary'):
                    # Available since OpenAPI 3.1.0, FastAPI 0.99.0
                    openapi_args['summary'] = self._app.summary
                return fastapi.openapi.utils.get_openapi(**openapi_args)
```
But mypy didn't like that:
```
fastapi_versionizer/versionizer.py:231: error: Argument 1 to "get_openapi" has incompatible type "**dict[str, Sequence[object] | dict[str, str | Any] | None]"; expected "str"  [arg-type]
fastapi_versionizer/versionizer.py:231: error: Argument 1 to "get_openapi" has incompatible type "**dict[str, Sequence[object] | dict[str, str | Any] | None]"; expected "str | None"  [arg-type]
fastapi_versionizer/versionizer.py:231: error: Argument 1 to "get_openapi" has incompatible type "**dict[str, Sequence[object] | dict[str, str | Any] | None]"; expected "Sequence[BaseRoute]"  [arg-type]
fastapi_versionizer/versionizer.py:231: error: Argument 1 to "get_openapi" has incompatible type "**dict[str, Sequence[object] | dict[str, str | Any] | None]"; expected "Sequence[BaseRoute] | None"  [arg-type]
fastapi_versionizer/versionizer.py:231: error: Argument 1 to "get_openapi" has incompatible type "**dict[str, Sequence[object] | dict[str, str | Any] | None]"; expected "list[dict[str, Any]] | None"  [arg-type]
fastapi_versionizer/versionizer.py:231: error: Argument 1 to "get_openapi" has incompatible type "**dict[str, Sequence[object] | dict[str, str | Any] | None]"; expected "list[dict[str, str | Any]] | None"  [arg-type]
fastapi_versionizer/versionizer.py:231: error: Argument 1 to "get_openapi" has incompatible type "**dict[str, Sequence[object] | dict[str, str | Any] | None]"; expected "dict[str, str | Any] | None"  [arg-type]
fastapi_versionizer/versionizer.py:231: note: "Dict" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
fastapi_versionizer/versionizer.py:231: note: Consider using "Mapping" instead, which is covariant in the value type
fastapi_versionizer/versionizer.py:231: error: Argument 1 to "get_openapi" has incompatible type "**dict[str, Sequence[object] | dict[str, str | Any] | None]"; expected "bool"  [arg-type]
Found 8 errors in 1 file (checked 9 source files)
```
So this solution is a bit easier. I hope you still like it. 😃

### CODE REVIEW

File: fastapi_versionizer/versionizer.py [hunk 1]
```diff
@@ -216,11 +216,28 @@ def _add_version_docs(
         if self._include_version_openapi_route and self._app.openapi_url is not None:
             @router.get(self._app.openapi_url, include_in_schema=False)
             async def get_openapi() -> Any:
-                return fastapi.openapi.utils.get_openapi(
-                    title=title,
-                    version=version_str,
-                    routes=router.routes
-                )
+                if hasattr(self._app, 'summary'):
```
Comment (alexschimpf) [inline fastapi_versionizer/versionizer.py:219]: You could do this without as much redundancy by doing something like this:
```
openapi_params = {
    'title': title,
    'version': version_str,
    'description': self._app.description,
    'routes': router.routes,
    'terms_of_service': self._app.terms_of_service,
    'contact': self._app.contact,
    'license_info': self._app.license_info
}

if hasattr(self._app, 'summary'):
    openapi_params['summary'] = self._app.summary

return fastapi.openapi.utils.get_openapi(**openapi_params)
```

Comment (alexschimpf) [reply fastapi_versionizer/versionizer.py:219]: Oh woops, I just read your PR description. Hmm lemme look into that.

Comment (alexschimpf) [reply fastapi_versionizer/versionizer.py:219]: Oh okay, all you have to do is add a type for the `openapi_params` (or `openapi_args`). So do like this:
```
openapi_params: Dict[str, Any] = {
    'title': title,
    'version': version_str,
    'description': self._app.description,
    'routes': router.routes,
    'terms_of_service': self._app.terms_of_service,
    'contact': self._app.contact,
    'license_info': self._app.license_info
}

if hasattr(self._app, 'summary'):
    openapi_params['summary'] = self._app.summary

return fastapi.openapi.utils.get_openapi(**openapi_params)
```

Comment (worldworm) [reply fastapi_versionizer/versionizer.py:229]: Oh my, I didn't think the solution would be so easy to keep mypy happy 😴 
Thanks for the hint 😄

**Comment by alexschimpf** [comment]
@worldworm Sorry, one last request. Can you update the "simple" test/example to include some of these fields so we can make sure this case has proper test coverage?

**Comment by worldworm** [comment]
Sure, that's a good idea. 
I have added description and termsOfService. I think they look better than a license_info or your name as a contact. 😄 
I thought about checking the summary attribute, but having a version dependency in the test case doesn't look "simple.py" to me. Hope you are fine with my pick.

**Pull request merged**